### ISSUE TITLE
Protect Twitch OAuth state

### ISSUE THREAD

**Issue by Perdolique** [Perdolique/perd#103]
## Context

This is the Twitch OAuth security prerequisite for account linking in #716. The current authorization request places only `redirectTo` in `state`, so the callback has no server-verifiable, single-use correlation value.

## Outcome

Every Twitch sign-in or linking attempt uses a server-verifiable, expiring, single-use OAuth state value that preserves only a safe application redirect and the intended operation.

## Behavior and contracts

- Generate a cryptographically random state nonce before redirecting to Twitch.
- Store only the data required to validate the nonce, its expiry, the safe internal redirect, and whether the intent is sign-in or account linking.
- Bind a linking attempt to the initiating application user and session.
- Validate and consume state before exchanging the authorization code.
- Reject missing, malformed, mismatched, expired, or replayed state without calling Twitch's token or user APIs.
- Preserve the safe redirect after successful sign-in or linking.
- Do not treat a raw redirect target as OAuth state.
- Keep raw provider failures in telemetry while returning safe user-facing authentication errors.

## Acceptance criteria

- Valid sign-in and linking callbacks each consume one matching state value and continue with their intended operation.
- A callback cannot change from sign-in to linking or target another user.
- Missing, invalid, expired, and replayed state performs no Twitch token exchange and creates or links no identity.
- External, protocol-relative, and otherwise unsafe redirect targets cannot be persisted or followed.
- Concurrent replay attempts allow at most one successful consumption.
- Focused tests cover issuance, expiry, replay, intent binding, user/session binding, safe redirects, and side-effect ordering.

## Dependencies / Out of scope

Part of #716. This must be complete before #104.

Turnstile, email authentication, Twitch account linking behavior, additional OAuth providers, and account merging are out of scope.
